### PR TITLE
fix: apply updated spec for usage_unit to aws cost trend widget and update widget data table

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
Commit messages say it all.
image
image

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue [hunk 1]
```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
...
-            type: state.fieldsKey === 'cost_sum' ? 'cost' : 'number',
+        const _textOptions: Field['textOptions'] = state.fieldsKey === 'cost' ? {
+            type: 'cost',
+        } : {
+            type: 'usage',
+            unitPath: 'usage_unit',
         };
 
-        const refinedFields = getWidgetTableDateFields(widgetState.granularity, widgetState.dateRange, _textOptions, state.fieldsKey);
+        const refinedFields = getWidgetTableDateFields(widgetState.granularity, widgetState.dateRange, _textOptions, 'value_sum');
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:134]: note: `unitPath` is the path to get the value of unit when refining value of each cell.

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue [hunk 2]
```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
...
-            ['cost_sum', 'usage_quantity_sum'],
-        );
+            ['value_sum'],
+        ).map((data) => {
+            let value = data[USAGE_TYPE_VALUE_KEY] ?? 'Unknown';
+            if (state.fieldsKey === 'usage_quantity') value = `${value}${data.usage_unit ? ` (${data.usage_unit})` : ''}`;
+            return {
+                ...data,
+                [USAGE_TYPE_VALUE_KEY]: value,
+            };
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:161]: note: the first cell data (USAGE_TYPE_VALUE_KEY data) must be refined.
If that is empty, must be converted to 'Unknown'.
And usage_unit must be displayed together.

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue [hunk 3]
```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
...
         return tableData;
     }),
-    legends: computed<Legend[]>(() => getXYChartLegends(state.data?.results, USAGE_TYPE_VALUE_KEY)),
+    legends: computed<Legend[]>(() => {
+        const data = state.data?.results ?? [];
+        const legends: Legend[] = getXYChartLegends(data, USAGE_TYPE_VALUE_KEY)
+            .map((l, i) => {
+                let label = l.label;
+                if (state.fieldsKey === 'usage_quantity') label = `${label}${data[i]?.usage_unit ? ` (${data[i]?.usage_unit})` : ''}`;
+                return { ...l, label };
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:172]: note: note: the legend data must be refined as the same with the table data.
If that is empty, must be converted to 'Unknown'. -> `getXYChartLegends` helper does it.
And `usage_unit` must be displayed together.

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue [hunk 4]
```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
...
+        return legends;
+    }),
     showLegendsOnTable: computed(() => widgetState.options.legend_options?.enabled && widgetState.options.legend_options.show_at === 'table'),
     chart: null as XYChart | null,
+    showChart: computed(() => {
+        if (state.fieldsKey === 'cost') return true;
+        if (!state.data?.results) return true;
+        // hide chart when there are different usage_unit in data
+        return uniqBy(state.data.results, 'usage_unit').length === 1;
+    }),
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:183]: note: chart must be hidden if there are different value of usage_unit in each array data.

File: apps/web/src/services/dashboards/widgets/_components/WidgetDataTable.vue [hunk 5]
```diff
@@ -95,40 +94,27 @@ const textFormatter = (value:string|number, textOptions: Field['textOptions']) =
...
     } if (textOptions?.type === 'percent') {
         return `${value.toFixed(1)}%`;
     }
+    if (textOptions?.type === 'usage') {
+        let unit = textOptions?.unit;
+        const unitPath = textOptions?.unitPath;
+        if (!unit && unitPath) unit = getValueByPath(cellData, unitPath);
+        // TODO: change to use usageFormatter
+        return numberFormatter(value) + (unit ? ` ${unit}` : '');
+    }
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_components/WidgetDataTable.vue:110]: praise: 👍

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue [hunk 6]
```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
 const state = reactive({
     loading: true,
     data: null as FullData | null,
-    fieldsKey: computed<'cost_sum'|'usage_quantity_sum'>(() => (selectedSelectorType.value === 'cost' ? 'cost_sum' : 'usage_quantity_sum')),
+    fieldsKey: computed<'cost'|'usage_quantity'>(() => (selectedSelectorType.value === 'cost' ? 'cost' : 'usage_quantity')),
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:114]: praise: oh, I'll try this too

File: apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue (hunk 4 above)
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue:183]: I'll copy this🫡

**Review by yuda110 — APPROVED** [review]
LGTM!!!

**Review by yuda110 — APPROVED** [review]

**Pull request merged**